components/AdminUserSettings/AdminUserSettings.test.tsx

```diff
@@ -1,12 +1,104 @@
 import { AdminUserSettings } from './AdminUserSettings';
-import { render, screen } from '@testing-library/react';
+import {
+  fireEvent,
+  getByTestId,
+  render,
+  screen,
+  waitFor,
+} from '@testing-library/react';
 import React from 'react';
 
+const mockPush = jest.fn();
+const mockUsePathname = jest.fn();
+const mockLogoutAccount = jest.fn();
+
+const mockUseAuthContext = {
+  logoutAccount: mockLogoutAccount,
+};
+
+jest.mock('../../context/AuthContextProvider', () => ({
+  useAuthContext() {
+    return {
+      ...mockUseAuthContext,
+    };
+  },
+}));
+
+jest.mock('next/navigation', () => ({
+  useRouter() {
+    return {
+      push: mockPush,
+    };
+  },
+  usePathname() {
+    return mockUsePathname();
+  },
+}));
+
 describe('AdminUserSettings Component', () => {
-  it('should render the component', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+
     render(<AdminUserSettings />);
+  });
+
+  it('should render the component', async () => {
     const adminUserSettings = screen.getByTestId('admin-user-settings');
 
     expect(adminUserSettings).toBeInTheDocument();
   });
+
+  it('should show user options when clicked', () => {
+    const adminUserSettings = screen.getByTestId('admin-user-settings');
+
+    fireEvent.click(adminUserSettings);
+
+    waitFor(() => {
+      expect(screen.getByTestId('edit-profile-button')).toBeInTheDocument();
+      expect(screen.getByTestId('sign-out-button')).toBeInTheDocument();
+    });
+  });
+
+  it('should not show user options when closed', () => {
+    const adminUserSettings = screen.getByTestId('admin-user-settings');
+
+    fireEvent.click(adminUserSettings);
+
+    fireEvent.click(adminUserSettings);
+
+    waitFor(() => {
+      expect(screen.getByTestId('edit-profile-button')).not.toBeInTheDocument();
+      expect(screen.getByTestId('sign-out-button')).not.toBeInTheDocument();
+    });
+  });
+
+  it('should direct to /admin/edit-profile route when the edit profile button is clicked', () => {
+    const adminUserSettings = screen.getByTestId('admin-user-settings');
+
+    fireEvent.click(adminUserSettings);
+
+    waitFor(() => {
+      const editProfileButton = screen.getByTestId('edit-profile-button');
+      fireEvent.click(editProfileButton);
```

Comment:
nit: we could just directly place the `editProfileButton` into the `fireEvent.click` as an argument instead of declaring a variable

Reply:
we could, but I don't think I've done this in previous PRs